yt_dlp/extractor/iwara.py

```diff
@@ -63,20 +65,27 @@ def _extract_formats(self, video_id, fileurl):
         # https://github.com/yt-dlp/yt-dlp/issues/6549#issuecomment-1473771047
         x_version = hashlib.sha1('_'.join((paths[-1], q['expires'][0], '5nFp9kmbNnHdAFhaqMvt')).encode()).hexdigest()
 
+        preference = qualities([
+            'preview', '360', '540', 'Source'
+        ])
+
         files = self._download_json(fileurl, video_id, headers={'X-Version': x_version})
         for fmt in files:
             yield traverse_obj(fmt, {
                 'format_id': 'name',
                 'url': ('src', ('view', 'download'), {self._proto_relative_url}),
                 'ext': ('type', {mimetype2ext}),
-                'quality': ('name', {lambda x: int_or_none(x) or 1e4}),
+                'quality': ('name', {lambda x: preference(x)}),
                 'height': ('name', {int_or_none}),
             }, get_all=False)
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         video_data = self._download_json(f'http://api.iwara.tv/video/{video_id}', video_id)
 
+        if video_data.get('fileUrl') is None:
+            raise ExtractorError('No fileUrl attribute, file is unplayable')
```

Comment:
any suggestions to make this message better?

Reply:
Is this an expected error? Example?

Reply:
confirmed on my own, but:
- `if not video_data.get('fileUrl'):` is enough for this predicate
- The returned JSON seems to have `embedUrl` that points to a YouTube video (https://youtu.be/wv53vrRx3aY), the extractor should redirect to this URL instead

Reply:
```suggestion
            if video_data.get('embedUrl'):
                return self.url_result(video_data.get('embedUrl'))
            raise ExtractorError('This video is unplayable', expected=True)
```
Can't test it now because Iwara times out very frequently now